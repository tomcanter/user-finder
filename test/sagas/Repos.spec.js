import {
  getRepos,
} from 'store/sagas/Repos';

describe('Saga: getRepos', () => {

  describe('when requesting repos', () => {
    it('should call the API with the url and parameters', () => {
      const generator = getRepos({payload: {url: 'repos.net'}});
      const repos = generator.next().value;
      expect(repos.CALL).toMatchSnapshot();
    });

    it('should put a success action with the normalized data', () => {
      const generator = getRepos({payload: {url: 'repos.net'}});
      const response = {
        data: [
          {id: 1, name: 'test'},
          {id: 2, name: 'hello'},
        ],
      };
      generator.next();
      const normalized = generator.next(response).value;
      expect(normalized.PUT).toMatchSnapshot();
    });
  });

  describe('when a request to the API fails', () => {
    it('should call a failure action', () => {
      const generator = getRepos({payload: {url: 'repos.net'}});
      const apiError = new Error('it went wrong');
      apiError.response = 'test';

      generator.next();
      const errorAction = generator.throw(apiError).value;
      expect(errorAction.PUT).toMatchSnapshot();
    });
  });

});
