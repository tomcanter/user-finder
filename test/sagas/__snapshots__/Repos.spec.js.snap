// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Saga: getRepos when a request to the API fails should call a failure action 1`] = `
Object {
  "action": Object {
    "message": "it went wrong",
    "response": "test",
    "type": "REPOS_FAILURE",
  },
  "channel": null,
}
`;

exports[`Saga: getRepos when requesting repos should call the API with the url and parameters 1`] = `
Object {
  "args": Array [
    "repos.net",
    Object {
      "params": Object {
        "per_page": 10,
        "sort": "pushed",
      },
    },
  ],
  "context": null,
  "fn": [Function],
}
`;

exports[`Saga: getRepos when requesting repos should put a success action with the normalized data 1`] = `
Object {
  "action": Object {
    "entities": Object {
      "repos": Object {
        "1": Object {
          "id": 1,
          "name": "test",
        },
        "2": Object {
          "id": 2,
          "name": "hello",
        },
      },
    },
    "result": Array [
      1,
      2,
    ],
    "type": "REPOS_SUCCESS",
    "url": "repos.net",
  },
  "channel": null,
}
`;
